lib/enigma.rb

```diff
@@ -1,71 +1,106 @@
 require 'date'
 class Enigma
-  def rand_key
-    rand(99999)
-  end
+  attr_reader :alphabet
 
-  def zero_pad(key)
-    pad_num = key.to_s
-    pad_num.insert(0, "0") until pad_num.size == 5
-    pad_num
+  def initialize
+    @alphabet = Array("a".."z")<< " "
   end
 
-  def shift_keys(key)
-    keys = []
-    key.chars.each_with_index do |char, index|
-      next if index == 4
-      keys << char + key[index + 1]
+  def cipher(string, shifts)
+    indexes_plus_shifts = []
+    to_indexes(string).each_with_index do |element, index|
+      if element.class == String
+        indexes_plus_shifts << element
+      else
+        indexes_plus_shifts << element + shifts[index%4]
+      end
     end
-    keys
-  end
-
-  def date
-    date = Date.today.strftime("%d%m%y")
-  end
-
-  def produce_offset(date)
-    squared_date = date.to_i * date.to_i
-    squared_date.to_s.chars.last(4)
+    indexes_to_string(indexes_plus_shifts)
   end
 
-  def shifts(key, date)
-    key_array = shift_keys(zero_pad(key))
-    offset = produce_offset(date)
-    shifts = []
-    offset.each_with_index do |element, index|
-      shifts << element.to_i + key_array[index].to_i
+  def decipher(string, shifts)
+    indexes_plus_shifts = []
+    to_indexes(string).each_with_index do |element, index|
+      if element.class == String
+        indexes_plus_shifts << element
+      else
+        indexes_plus_shifts << element - shifts[index%4]
+      end
     end
-    shifts
+    indexes_to_string(indexes_plus_shifts)
   end
 
   def to_indexes(string)
-    alphabet = Array("a".."z")<< " "
     string_array = string.downcase.chars
     string_array.map do |char|
-      alphabet.find_index(char)
+      if @alphabet.include?(char) == false
+        char = char
+      else
+        @alphabet.find_index(char)
+      end
     end
   end
 
-  def indexes_to_string(string, shifts)
-    alphabet = Array("a".."z")<< " "
+  def indexes_to_string(shifts)
     shifts.map do |number|
-      alphabet.rotate(number)[0]
+      if number.class == String
+        number
+      else
+        @alphabet.rotate(number)[0]
+      end
     end.join
   end
 
-  def cipher(string, shifts)
-    indexes_plus_shifts = []
-    to_indexes(string).each_with_index do |number, index|
-      indexes_plus_shifts << number + shifts[index%4]
-    end
-    indexes_to_string(string, indexes_plus_shifts)
+  def encryption_output(string, key, date)
+    output = {}
+    output[:encryption] = string
+    output[:key] = key
+    output[:date] = date
+    output
   end
```

Comment:
Added the `encryption_output` and `decryption_output` methods to array the outputs as a hash